graylog2-server/src/main/java/org/graylog2/alarmcallbacks/EmailAlarmCallback.java

```diff
@@ -106,6 +95,19 @@ public void call(Stream stream, AlertCondition.CheckResult result) {
         }
     }
 
+    protected List<Message> getAlarmBacklog(AlertCondition.CheckResult result) {
+        final AlertCondition alertCondition = result.getTriggeredCondition();
+        final List<MessageSummary> backlogSummaries = result.getMatchingMessages()
+                .subList(0, Math.min(alertCondition.getBacklogSize(), result.getMatchingMessages().size()));
+        final List<Message> backlog = Lists.newArrayList();
```

Comment:
We already know `backlogSummaries.size()`, so we can initialize the list with the correct size.